### PR TITLE
Create a 'SeededFileStore' cache backend

### PR DESCRIPTION
**Pull request by ahal**
This will allow us to setup cache configurations that can automatically download and extract the bugbug adr cache as seed. It can be set up like this:
```
[adr.cache.stores]
file = {
    driver = "seeded-file",
    path = "/home/ahal/tmp/adr_cache",
    url = "https://s3-us-west-2.amazonaws.com/communitytc-bugbug/data/adr_cache.tar.zst",
    archive_relpath = "data/adr_cache",
    reseed_interval = 10080
}
```

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -103,6 +103,25 @@ For example:
...
+    file = {
+        driver = "seeded-file",
+        path = "/path/to/dir/to/keep/cache",
+        url = "https://example.com/adr_cache.tar.gz"
+    }
+
+Supported archive formats include ``.zip``, ``.tar``, ``.tar.gz``, ``.tar.bz2`` and ``.tar.zst``.
+
+The config also accepts a ``reseed_interval`` (in minutes) which will re-seed the cache after the
+interval expires. This assumes the URL is automatically updated by some other process.
```
Comment (marco-c) [inline docs/usage.rst]: Note: you could also use the eTag to check if there is a new version.

File: test/test_cache_stores.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    def callback(resp):
+        resp.callback_processd = True
+
+        name = resp.url.split('/')[-1]
+        fh = open(archive_dir / name, 'rb')
+        resp.raw = fh
+
+        return resp
+
+    with responses.RequestsMock(response_callback=callback) as rsps:
```
Comment (marco-c) [inline test/test_cache_stores.py]: Nit: you can use assert_all_requests_are_fired=True to make sure all mock responses you add are fired

File: adr/util/cache_stores.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        """Download and extract the seed data to the cache directory."""
+        logger.info(f"Seeding adr cache at {self._directory} with contents from {self._url}")
+        self._create_cache_directory(self._directory)
+
+        filename = self._url.split('/')[-1]
+        with self._session.get(self._url, stream=True) as r:
+            r.raise_for_status()
+
+            try:
+                tempdir = tempfile.mkdtemp()
```
Comment (marco-c) [inline adr/util/cache_stores.py]: Nit: you can use TemporaryDirectory to avoid the try-finally

File: docs/usage.rst (hunk 1 above)
Comment (ahal) [reply docs/usage.rst]: Ok, maybe that could be implemented in a follow-up.

File: test/test_cache_stores.py (hunk 2 above)
Comment (ahal) [reply test/test_cache_stores.py]: According to the docs, that is the default behaviour (and is inline with my experience running the tests as well).

File: adr/util/cache_stores.py (hunk 3 above)
Comment (ahal) [reply adr/util/cache_stores.py]: Done.

**Pull request merged**